cdap-api/src/main/java/co/cask/cdap/api/flow/FlowletConnection.java

```diff
@@ -49,6 +55,14 @@ public Type getSourceType() {
   }
 
   /**
+   * @return Type of source.
+   */
+  public String getSourceNamespace() {
```

Comment:
If you decide not to do the StreamFlowletConnection then specifically document when this will return null.